crates/sdk/src/prover/agg.rs

```diff
@@ -146,28 +159,208 @@ impl<E: StarkFriEngine<SC>> AggStarkProver<E> {
+
+                        for (idx, proof_input, parent_span) in proof_input_rx.iter() {
+                            let span =
+                                parent_span.in_scope(|| info_span!("prove_segment", idx = idx));
+                            let _guard = span.enter();
+
+                            // Generate proof
+                            let proof = vm.prove_single(&vm_pk.vm_pk, proof_input);
+
+                            // Send proof
+                            proof_tx.send((idx, proof)).expect("Failed to send proof");
+                        }
+
+                        drop(proof_tx);
+                    });
+
+                    // ===== Collector Thread =====
+                    let collector_handle = s.spawn(move || {
+                        let mut proofs = Vec::with_capacity(internal_inputs_num);
+                        let mut pending_proofs = std::collections::BTreeMap::new();
+                        let mut next_idx = 0;
+
+                        for (idx, proof) in proof_rx.iter() {
+                            pending_proofs.insert(idx, proof);
+
+                            // Collect proofs in order
+                            while let Some(proof) = pending_proofs.remove(&next_idx) {
+                                proofs.push(proof);
+                                next_idx += 1;
+                            }
+                        }
+
+                        // Ensure all proofs have been collected in order
+                        let expected = internal_inputs_num;
+                        let actual = proofs.len();
+                        assert_eq!(
+                            actual, expected,
+                            "Expected {} proofs, but got {}",
+                            expected, actual
+                        );
+
+                        proofs
+                    });
+
+                    // ===== Main Thread - Send Inputs =====
+                    for (idx, input) in internal_inputs.into_iter().enumerate() {
+                        internal_node_idx += 1;
+                        let span = info_span!("single_internal_agg", idx = internal_node_idx);
+                        let _guard = span.enter();
+
+                        // Send input
+                        input_tx
+                            .send((idx, input.write().into(), span.clone()))
+                            .expect("Failed to send internal node input");
+                    }
+
+                    drop(input_tx);
+
+                    // Wait for all threads to complete
+                    trace_handle.join().expect("Trace thread panicked");
+                    prove_handle.join().expect("Prove thread panicked");
+
+                    collector_handle.join().expect("Collector thread panicked")
+                });
+
+                tracing::info!(
+                    "agg_layer wall time, group = internal.{internal_node_height}: {:.3}s",
```

Comment:
the group should already be in the outer span

Reply:
I removed it entirely by changing the tracing span's timer to be the same as the wall clock